### PR TITLE
Code refactoring in preparation of HtmlCleaner replacement

### PR DESCRIPTION
**Pull request by cketti**

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/message/IdentityHeaderBuilder.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import com.fsck.k9.K9;
 import com.fsck.k9.activity.MessageReference;
 import com.fsck.k9.mail.internet.TextBody;
+import com.fsck.k9.message.quote.InsertableHtmlContent;
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/IdentityHeaderBuilder.java]: Probably an unused import.

File: k9mail/src/main/java/com/fsck/k9/message/MessageBuilder.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 import com.fsck.k9.mail.internet.MimeUtility;
 import com.fsck.k9.mail.internet.TextBody;
 import com.fsck.k9.mailstore.TempFileBody;
+import com.fsck.k9.message.quote.InsertableHtmlContent;
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/MessageBuilder.java]: Probably unused

File: k9mail/src/main/java/com/fsck/k9/message/TextBodyBuilder.java [hunk 3]
```diff
@@ -5,9 +5,10 @@
 import android.util.Log;
 
 import com.fsck.k9.K9;
-import com.fsck.k9.helper.HtmlConverter;
+import com.fsck.k9.message.html.HtmlConverter;
 import com.fsck.k9.mail.Body;
 import com.fsck.k9.mail.internet.TextBody;
+import com.fsck.k9.message.quote.InsertableHtmlContent;
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/TextBodyBuilder.java]: Probably unused.

File: k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            if (originalMessage.getSubject() != null) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_subject)).append(" ").append(originalMessage.getSubject()).append("\r\n");
+            }
+            quotedText.append("\r\n");
+
+            quotedText.append(body);
+
+            return quotedText.toString();
+        } else {
+            // Shouldn't ever happen.
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java]: Then should we throw an error?

File: k9mail/src/test/java/com/fsck/k9/message/PgpMessageBuilderTest.java [hunk 5]
```diff
@@ -33,6 +33,7 @@
 import com.fsck.k9.mail.internet.MimeUtility;
 import com.fsck.k9.mail.internet.TextBody;
 import com.fsck.k9.message.MessageBuilder.Callback;
+import com.fsck.k9.message.quote.InsertableHtmlContent;
```
Comment (philipwhiuk) [inline k9mail/src/test/java/com/fsck/k9/message/PgpMessageBuilderTest.java]: Unused.

File: k9mail/src/test/java/com/fsck/k9/message/TextBodyBuilderTest.java [hunk 6]
```diff
@@ -3,6 +3,7 @@
 import com.fsck.k9.Account.QuoteStyle;
 import com.fsck.k9.mail.internet.TextBody;
 
+import com.fsck.k9.message.quote.InsertableHtmlContent;
```
Comment (philipwhiuk) [inline k9mail/src/test/java/com/fsck/k9/message/TextBodyBuilderTest.java]: Unused

File: k9mail/src/test/java/com/fsck/k9/message/html/HtmlSanitizerTest.java [hunk 7]
```diff
@@ -1,6 +1,7 @@
-package com.fsck.k9.helper;
+package com.fsck.k9.message.html;
 
 
+import com.fsck.k9.message.html.HtmlSanitizer;
```
Comment (philipwhiuk) [inline k9mail/src/test/java/com/fsck/k9/message/html/HtmlSanitizerTest.java]: This shouldn't be necessary - it's in the same package.

File: k9mail/src/main/java/com/fsck/k9/message/quote/QuoteHelper.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    static String getSentDateText(Resources resources, Message message) {
+        try {
+            final int dateStyle = DateFormat.LONG;
+            final int timeStyle = DateFormat.LONG;
+            Date date = message.getSentDate();
+            Locale locale = resources.getConfiguration().locale;
+            return DateFormat.getDateTimeInstance(dateStyle, timeStyle, locale)
+                    .format(date);
+        } catch (Exception e) {
+            return "";
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/quote/QuoteHelper.java]: Is it worth logging here?

File: k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * Add quoting markup to a text message.
+     * @param originalMessage Metadata for message being quoted.
+     * @param messageBody Text of the message to be quoted.
+     * @param quoteStyle Style of quoting.
+     * @return Quoted text.
+     * @throws MessagingException
+     */
+    public static String quoteOriginalTextMessage(Resources resources, Message originalMessage, String messageBody, QuoteStyle quoteStyle, String prefix) throws MessagingException {
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java]: Line is a bit long

File: k9mail/src/main/java/com/fsck/k9/message/signature/HtmlSignatureRemover.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+            } else {
+                // No blockquotes found.
+                content = content.substring(0, dashSignatureHtml.start());
+            }
+        }
+
+        // Fix the stripping off of closing tags if a signature was stripped,
+        // as well as clean up the HTML of the quoted message.
+        HtmlCleaner cleaner = new HtmlCleaner();
+        CleanerProperties properties = cleaner.getProperties();
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/message/signature/HtmlSignatureRemover.java]: Haven't we got a set of HTMLCleaner properties we define elsewhere. Shouldn't they be the same?

(Not a big deal now we're getting rid of it of course).

File: k9mail/src/main/java/com/fsck/k9/message/IdentityHeaderBuilder.java (hunk 1 above)
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/message/IdentityHeaderBuilder.java]: The class was moved out of this package which is why now an import statement is necessary. Same for the other occurrences.

File: k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java (hunk 9 above)
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java]: None of the code was changed. It was only moved around. Cleaning this is up and adding tests is a separate task.

**Comment by cketti** [comment]
I should have mentioned that I made no functional changes. This is just moving stuff around.

File: k9mail/src/test/java/com/fsck/k9/message/html/HtmlSanitizerTest.java (hunk 7 above)
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/message/html/HtmlSanitizerTest.java]: :+1:

**Comment by philipwhiuk** [comment]
That's fine. I don't mind if we pick that stuff up in another PR.

**Pull request merged**